contracts/LockedFund/LockedFund.sol

```diff
@@ -394,72 +518,118 @@ contract LockedFund is ILockedFund {
 	 * @param _duration The duration for vesting.
 	 * @param _basisPoint The % (in Basis Point) which determines how much will be (waited) unlocked immediately.
 	 * @param _unlockedOrWaited Determines if the Basis Point determines the Unlocked or Waited Unlock Balance.
+	 * @param _receiveTokens - True if tokens should be taken from caller, False otherwise.
 	 */
 	function _depositVested(
 		address _userAddress,
```

Comment:
consider using safeTranserFrom from oz library here and further on, it might make difference if the token already exists

Reply:
Just to give a little context, in some cases, there will be token transfer directly to the LockedFund after the sale:
- Bonding Curve Tokens
- Cases where the initial fundraise determines the total (initial) token supply

So, we use the `_receiveTokens` parameter in those cases.

Reply:
sorry my comments sometimes slipped when commenting from IDE
what if tokens throw or return no value?
these cases are covered with 
`https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/utils/SafeERC20.sol` and `safeTransferFrom`
so suggested to use instead of 
```
if (_receiveTokens) {
			bool txStatus = token.transferFrom(msg.sender, address(this), _amount);
```

Reply:
For now, we are using the `txStatus` to check if the `transferFrom` is successful or not. That is checked below the one you quoted. The problem is that we cannot assume all tokens to implement the `safeTransferFrom` function (depending on each project which comes to Origins.)

Reply:
`SafeERC20.safeTransferFrom` relies on usual `ERC20.transferFrom` internally, it just correctly processes return value, including the case when there is none.
Using safeTransfer, safeTransferFrom, and approve (not safeApprove!) is the best approach in 99% of the cases.

Reply:
I understand the importance, but the problem is based on how the token might be implemented by the projects(I am not talking about Origins Token, but in general, for any token of a project, who might be interested in listing on Origins).

If they have not implemented SafeERC20 functions, our interface won't be compatible with their ERC20.

Please do let me know if that is not the case, or there might be a simple workaround.

Reply:
it's not the case.
`SafeERC20(token).safeTransfer(...)` uses some magic to actually call `ERC20(token).transfer(...)` internally. 
Thanks to that, any contract that implements `transfer(address,uint256)` function can be called via `SafeERC20.safeTransfer` wrapper.

Reply:
Solved in 9a40a24 and 0f1b7d8